freqtrade/optimize/hyperopt_interface.py

```diff
@@ -76,30 +81,102 @@ def generate_roi_table(params: Dict) -> Dict[int, float]:
         return roi_table
 
     @staticmethod
-    def stoploss_space() -> List[Dimension]:
+    def roi_space() -> List[Dimension]:
         """
-        Create a stoploss space.
+        Create a ROI space.
+
+        Defines values to search for each ROI steps.
+
+        This method implements adaptive roi hyperspace with varied
+        ranges for parameters which automatically adapts to the
+        ticker interval used.
+
+        It's used by Freqtrade by default, if no custom roi_space method is defined.
+        """
+
+        # Default scaling coefficients for the roi hyperspace. Can be changed
+        # to adjust resulting ranges of the ROI tables.
+        # Increase if you need wider ranges in the roi hyperspace, decrease if shorter
+        # ranges are needed.
+        roi_t_alpha = 1.0
+        roi_p_alpha = 1.0
+
+        ticker_interval_mins = timeframe_to_minutes(IHyperOpt.ticker_interval)
+
+        # We define here limits for the ROI space parameters automagically adapted to the
+        # ticker_interval used by the bot:
+        #
+        # * 'roi_t' (limits for the time intervals in the ROI tables) components
+        #   are scaled linearly.
+        # * 'roi_p' (limits for the ROI value steps) components are scaled logarithmically.
+        #
+        # The scaling is designed so that it maps exactly to the legacy Freqtrade roi_space()
+        # method for the 5m ticker interval.
+        roi_t_scale = ticker_interval_mins / 5
+        roi_p_scale = math.log1p(ticker_interval_mins) / math.log1p(5)
+        roi_limits = {
+            'roi_t1_min': int(10 * roi_t_scale * roi_t_alpha),
+            'roi_t1_max': int(120 * roi_t_scale * roi_t_alpha),
+            'roi_t2_min': int(10 * roi_t_scale * roi_t_alpha),
+            'roi_t2_max': int(60 * roi_t_scale * roi_t_alpha),
+            'roi_t3_min': int(10 * roi_t_scale * roi_t_alpha),
+            'roi_t3_max': int(40 * roi_t_scale * roi_t_alpha),
+            'roi_p1_min': 0.01 * roi_p_scale * roi_p_alpha,
+            'roi_p1_max': 0.04 * roi_p_scale * roi_p_alpha,
+            'roi_p2_min': 0.01 * roi_p_scale * roi_p_alpha,
+            'roi_p2_max': 0.07 * roi_p_scale * roi_p_alpha,
+            'roi_p3_min': 0.01 * roi_p_scale * roi_p_alpha,
+            'roi_p3_max': 0.20 * roi_p_scale * roi_p_alpha,
+        }
+        logger.debug(f"Using roi space limits: {roi_limits}")
+        p = {
+            'roi_t1': roi_limits['roi_t1_min'],
+            'roi_t2': roi_limits['roi_t2_min'],
+            'roi_t3': roi_limits['roi_t3_min'],
+            'roi_p1': roi_limits['roi_p1_min'],
+            'roi_p2': roi_limits['roi_p2_min'],
+            'roi_p3': roi_limits['roi_p3_min'],
+        }
+        logger.info(f"Min roi table: {IHyperOpt.generate_roi_table(p)}")
+        p = {
```

Comment:
this assignment is just for printing / logging?

in that case, i would round the values to something sensible (5 maybe). i don't care about the 12th decimal being 5, and i doubt users will...

```
2019-08-15 20:29:09,633 - freqtrade.optimize.hyperopt_interface - INFO - Min roi table: {0: 0.03, 10: 0.02, 20: 0.01, 30: 0}
2019-08-15 20:29:09,633 - freqtrade.optimize.hyperopt_interface - INFO - Max roi table: {0: 0.31000000000000005, 40: 0.11000000000000001, 100: 0.04, 220: 0}

```

Reply:
yes, that's for logging, I'll consider rounding here

Reply:
The problem here is that we need to print the resulting table (returned by `generate_roi_table()`), we need to traverse all the items and round them somehow for printing... Have not found an elegant solution yet...

Reply:
`{k: round(v, 5) for k, v in roi_limits.items()}` ?

I think rounding would make sense for all ROI table outputs anyway - so it can be a generic solution within the generate_roi_table section?

Reply:
Thanks for the idea, seems to be working fine!
The generic `round_dict()` function was created in the misc.py
The details of hyperopt result (roi table, stoploss) are also rounded in the same way when printed in the human-readable format, not only logs. Please review the results.

I could also make the rounding more sophisticated (a generic function will be created and used), so that it rounds to n significant digits after the decimal point. (0.0012345678 will be rounded to 0.0012345 if n=5, now it rounds to 0.00123)

(Don not merge yet pls 😆  )

Reply:
An example of the current output (with rounding):

```
2019-08-20 23:39:08,884 - freqtrade.optimize.hyperopt_interface - DEBUG - Using roi space limits: {'roi_t1_min': 120, 'roi_t1_max': 1440, 'roi_t2_min': 120, 'roi_t2_max': 720, 'roi_t3_min': 120, 'roi_t3_max': 480, 'roi_p1_min': 0.022943223880069138, 'roi_p1_max': 0.09177289552027655, 'roi_p2_min': 0.022943223880069138, 'roi_p2_max': 0.160602567160484, 'roi_p3_min': 0.022943223880069138, 'roi_p3_max': 0.4588644776013828}
2019-08-20 23:39:08,884 - freqtrade.optimize.hyperopt_interface - INFO - Min roi table: {0: 0.06883, 120: 0.04589, 240: 0.02294, 360: 0}
2019-08-20 23:39:08,884 - freqtrade.optimize.hyperopt_interface - INFO - Max roi table: {0: 0.71124, 480: 0.25238, 1200: 0.09177, 2640: 0}

...

Best result:

...

ROI table:
{0: 0.61024, 257: 0.16258, 463: 0.02873, 1339: 0}
Stoploss: -0.14797
```

Reply:
> The values in the debug message are not rounded. 

that's perfectly fine ... if you look at debug, output is messy anyway :smile:

Reply:
> that's perfectly fine ... if you look at debug, output is messy anyway 😄

But it contains correct values. Which may have sense in case of debug.

Debug output is for people strong in spirit... 😄

Reply:
not sure if using hyperopt drunk will improve performance ... but ... 👍